js/join.js

```diff
@@ -0,0 +1,290 @@
+const funFact = document.getElementById('funFact');
+const next2 = document.getElementById('next2');
+const previous2 = document.getElementById('previous2');
+
+//variables for why RDS Page
+const whyRds = document.getElementById('whyRds');
+const heardAbout = document.getElementById('heardAbout');
+const previous3 = document.getElementById('previous3');
+const previewBtn = document.getElementById('next3');
+
+//variables for preview pages
+const previewFName = document.getElementById('previewFName');
+const previewLName = document.getElementById('previewLName');
+const previewCity = document.getElementById('previewCity');
+const previewState = document.getElementById('previewState');
+const previewCountry = document.getElementById('previewCountry');
+const previewIntro = document.getElementById('previewIntro');
+const previewSkills = document.getElementById('previewSkills');
+const previewInstitution = document.getElementById('previewInstitution');
+const previewFunFact = document.getElementById('previewFunFact');
+const previewForFun = document.getElementById('previewForFun');
+const previewWhyRds = document.getElementById('previewWhyRds');
+const previewHeardAbout = document.getElementById('previewHeardAbout');
+const previous4 = document.getElementById('previous4');
+const submit = document.getElementById('next4');
+
+//Vatiables for Completed page
+const personalLink = document.getElementById('personalLink');
+const copyBtn = document.getElementById('copy');
+
+function fetchSavedDetails() {
+  fetch('https://api.realdevsquad.com/users/self', {
+    headers: { 'content-type': 'application/json' },
+    credentials: 'include',
+  })
+    .then((res) => res.json())
+    .then((res) => {
+      window.localStorage.setItem('firstName', res.first_name);
+      window.localStorage.setItem('lastName', res.last_name);
+      console.log(res);
+      url = `https://api.realdevsquad.com/users/${res.id}/intro`;
+      personalLink.innerText = url;
+    });
+}
+
+const pages = [page1, page2, page3, page4, page5, page6];
+
+function selectPage() {
+  let currentFlowState = window.localStorage.getItem('flowState');
+  showPage(currentFlowState);
+}
+
+function showPage(currentFlowState) {
+  for (let i = 0; i < pages.length; i++) {
+    if (i == currentFlowState) {
+      pages[i].classList.remove('hide-page');
+    } else {
+      pages[i].classList.add('hide-page');
+    }
+  }
+}
+
+//Validators
+function arePersonalDetailsValid() {
+  return (
+    state.value.trim().length > 3 &&
+    city.value.trim().length > 3 &&
+    country.value.trim().length > 3
+  );
+}
+
+function introPageChecker() {
+  return (
+    introduction.value.trim().split(' ').length > 100 &&
+    skills.value.trim().split(' ').length > 6 &&
+    college.value.trim().split(' ').length > 5 &&
+    forFun.value.trim().split(' ').length > 100 &&
+    funFact.value.trim().split(' ').length > 100
+  );
+}
+
+function whyRdsPageChecker() {
```

Comment:
Can we please name the functions better? 
Instead of Checker word, we can use Validator